src/event/reader.rs

```diff
@@ -1038,6 +1104,18 @@ impl SliceMetadata {
     pub fn has_events(&self) -> bool {
         !self.partial_data_present && self.segment_data.value.len() > TYPE_PLUS_LENGTH_SIZE as usize
     }
+
+    fn copy_meta(&self) -> SliceMetadata {
```

Comment:
Maybe implement the clone trait?

Reply:
This method only copies the Segment metadata and not the segment Data Buffer. Hence I do not think we should implement the clone trait.